File: php/algorithms/cocktailSort.php
<?php
function cocktailSort($array)
{
    if (is_string($array))
        $array = str_split(preg_replace('/\s+/','',$array));

    do{
        $swapped = false;
        for($i=0;$i<count($array);$i++){
            if(isset($array[$i+1])){
                if($array[$i] > $array[$i+1]){
                    list($array[$i], $array[$i+1]) = array($array[$i+1], $array[$i]);
                    $swapped = true;
                }
            }
        }

        if ($swapped == false) break;

        $swapped = false;
        for($i=count($array)-1;$i>=0;$i--){
            if(isset($array[$i-1])){
                if($array[$i] < $array[$i-1]) {
                    list($array[$i],$array[$i-1]) = array($array[$i-1],$array[$i]);
                    $swapped = true;
                }
            }
        }
    }while($swapped);

    return $array;
}
$arrayTest = array(2, 1, 0, 10, -6, 12, 22);
echo "Unsorted Array\n";
echo implode(', ',$arrayTest );
echo "\nSorted Array\n";
echo implode(', ',cocktailSort($arrayTest)). PHP_EOL;

File: python/algorithms/CocktailSort.py
def cocktailSort(a): 
    n = len(a) 
    swapped = True
    start = 0
    end = n-1
    while (swapped==True): 
        swapped = False
        for i in range (start, end): 
            if (a[i] > a[i+1]) : 
                a[i], a[i+1]= a[i+1], a[i] 
                swapped=True
        if (swapped==False): 
            break
        swapped = False
        end = end-1
        for i in range(end-1, start-1,-1): 
            if (a[i] > a[i+1]): 
                a[i], a[i+1] = a[i+1], a[i] 
                swapped = True
        start = start+1
a = [5, 1, 4, 2, 8, 0, 2] 
cocktailSort(a) 
print("Sorted array is:") 
for i in range(len(a)): 
    print ("%d" %a[i]), 
